libs/designsystem/src/lib/components/button/button.component.spec.ts

```diff
@@ -463,4 +463,98 @@ describe('ButtonComponent', () => {
       });
     });
   });
+
+  const iconTestScenarios: { size: ButtonSize; iconPosition: 'left' | 'right'; expected: any }[] = [
+    {
+      size: ButtonSize.SM,
+      iconPosition: 'left',
+      expected: { paddingInline: '12px 16px' },
+    },
+    {
+      size: ButtonSize.SM,
+      iconPosition: 'right',
+      expected: { paddingInline: '16px 12px' },
+    },
+    {
+      size: ButtonSize.MD,
+      iconPosition: 'left',
+      expected: { paddingInline: '12px 16px' },
+    },
+    {
+      size: ButtonSize.MD,
+      iconPosition: 'right',
+      expected: { paddingInline: '16px 12px' },
+    },
+    {
+      size: ButtonSize.LG,
+      iconPosition: 'left',
+      expected: { paddingInline: '12px 16px' },
+    },
+    {
+      size: ButtonSize.LG,
+      iconPosition: 'right',
+      expected: { paddingInline: '16px 12px' },
+    },
+  ];
+  iconTestScenarios.forEach((scenario) => {
+    describe(`when configured with size = ${scenario.size}`, () => {
+      describe(`through one-time string initialization`, () => {
+        beforeEach(() => {
+          spectator = createHost(
+            `<button kirby-button size="${scenario.size}">
+              ${scenario.iconPosition === 'left' ? '<kirby-icon name="edit"></kirby-icon>' : ''}
+              <span>Text</span>
+              ${scenario.iconPosition === 'right' ? '<kirby-icon name="edit"></kirby-icon>' : ''}
+            </button>`
+          );
+          element = spectator.element as HTMLButtonElement;
+        });
+        it('should render with correct padding-inline', () => {
+          expect(element.getElementsByClassName('content-layer').length).toBe(1);
+          expect(element.getElementsByClassName('content-layer')[0]).toHaveComputedStyle({
+            'padding-inline': scenario.expected.paddingInline,
+          });
+        });
+      });
+
+      describe(`through an input property`, () => {
+        beforeEach(() => {
+          spectator = createHost(
+            `<button kirby-button size="${scenario.size}">
+              ${scenario.iconPosition === 'left' ? '<kirby-icon name="edit"></kirby-icon>' : ''}
+              <span>Text</span>
+              ${scenario.iconPosition === 'right' ? '<kirby-icon name="edit"></kirby-icon>' : ''}
+            </button>`
+          );
+          element = spectator.element as HTMLButtonElement;
+        });
+        it('should render with correct padding-inline', () => {
+          expect(element.getElementsByClassName('content-layer').length).toBe(1);
+          expect(element.getElementsByClassName('content-layer')[0]).toHaveComputedStyle({
+            'padding-inline': scenario.expected.paddingInline,
+          });
+        });
+      });
+
+      describe(`through template property binding`, () => {
```

Comment:
@thupi I have question that is probably stupid :) Isn't this `describe` block identical to the one above: `through an input property`?